lib/straight-server/orders_controller.rb

```diff
@@ -13,17 +15,26 @@ def initialize(env)
     end
 
     def create
-      
+
       unless @gateway
         StraightServer.logger.warn "Gateway not found"
         return [404, {}, "Gateway not found" ]
       end
 
+      unless @gateway.check_signature
+        ip = @env['HTTP_X_FORWARDED_FOR'].to_s
```

Comment:
By the way, if straight-server is used without a reverse-proxy, this line makes it vulnerable for spoofing.